lightning/src/ln/chan_utils.rs

```diff
@@ -550,6 +556,415 @@ pub fn build_htlc_transaction(prev_hash: &Txid, feerate_per_kw: u32, contest_del
 	}
 }
 
+/// Static channel fields used to build transactions given per-commitment fields (CommitmentTransactionInfo)
+#[derive(Clone)]
+pub struct ChannelStaticInfo {
+	/// Holder public keys
+	pub holder_pubkeys: ChannelPublicKeys,
+	/// Counter-party public keys
+	pub counterparty_pubkeys: ChannelPublicKeys,
+	/// The contest delay selected by the holder
+	pub holder_selected_contest_delay: u16,
+	/// The contest delay selected by the counterparty
+	pub counterparty_selected_contest_delay: u16,
+	/// The funding outpoint
+	pub funding_outpoint: OutPoint,
+	/// Whether the channel is outbound from the point of view of the holder
+	pub is_outbound_from_holder: bool,
```

Comment:
This is confusing, I would drop the _from_holder, and pass a holder bool in `get_channel_static_info` to mark the "direction" there.

Reply:
I don't understand the second part of the comment.

This field is just a copy of `Channel.is_outbound`.  The reason I expanded the name is because all the other fields have a party name (except `funding_outpoint` which is party independent), so I wanted to be consistent.

Reply:
Or maybe you are referring to the perspective argument `holder: bool` in the accessors `pubkeys()`, `contest_delay()` and `is_outbound()`.  Are you saying that instead of those arguments, the constructor should be modified to take into the account the perspective?

Reply:
Yes the issue is a naming nomenclature confusing between `holder/counterparty` and `broadcaster/countersignatory`. The first couple are fixed-point for the lifetime of the channels and ultimately always designate the same real entity. The second one are permutating in function of protocol phase and calling `Channel::build_commitment_transaction`, either for holder or counterparty txn.

Thus `ChannelStaticInfo` used in `build_commitment_transaction` should bind to the second nomenclature and `get_channel_static_info` extended with a boolean argument to assign `holder/counterparty` channel values either to `broadcaster/countersignatory` fields in function of direction.

Reply:
introduced `DirectedChannelStaticInfo` to address this, please take a look